R/revision_analysis.R

```diff
@@ -0,0 +1,209 @@
+      time_to = time_to_x_percent(lag, !!arg[[1]], percent = percent_final_value)
+    ) %>%
+    ungroup() %>%
+    mutate(
+      # TODO the units here may be a problem
+      min_lag = as.difftime(min_lag, units = "days"), # nolint: object_usage_linter
+      max_lag = as.difftime(max_lag, units = "days"), # nolint: object_usage_linter
+      time_to_pct_final = as.difftime(time_to, units = "days") # nolint: object_usage_linter
+    ) %>%
+    select(-time_to)
+  if (print_inform) {
+    total_num <- nrow(revision_behavior) # nolint: object_usage_linter
+    cli_inform("Number of revisions:")
+    cli_inform("Min lag (time to first version):")
+    difftime_summary(revision_behavior$min_lag)
+    if (!drop_nas) {
+      total_na <- nrow(epi_arch$DT %>% filter(is.na(!!arg[[1]]))) # nolint: object_usage_linter
+      cli_inform("Fraction of all versions that are `NA`:")
+      cli_li(num_percent(total_na, nrow(epi_arch$DT)))
+    }
+    total_num_unrevised <- sum(revision_behavior$n_revisions == 0) # nolint: object_usage_linter
+    cli_inform("No revisions:")
+    cli_li(num_percent(total_num_unrevised, total_num))
+    total_quickly_revised <- sum( # nolint: object_usage_linter
+      revision_behavior$max_lag <=
+        as.difftime(quick_revision, units = "days")
+    )
+    cli_inform("Quick revisions (last revision within {quick_revision}
+{units(quick_revision)} of the `time_value`):")
+    cli_li(num_percent(total_quickly_revised, total_num))
+    total_barely_revised <- sum( # nolint: object_usage_linter
+      revision_behavior$n_revisions <=
+        few_revisions
+    )
+    cli_inform("Few revisions (At most {few_revisions} revisions for that `time_value`):")
+    cli_li(num_percent(total_barely_revised, total_num))
+    cli_inform("")
+    cli_inform("Changes in Value:")
+
+    real_revisions <- revision_behavior %>% filter(n_revisions > 0) # nolint: object_usage_linter
+    n_real_revised <- nrow(real_revisions) # nolint: object_usage_linter
+    rel_change <- sum( # nolint: object_usage_linter
+      real_revisions$max_rel_change <
+        rel_change_threshold,
+      na.rm = TRUE
+    ) + sum(is.na(real_revisions$max_rel_change))
+    cli_inform("Less than {rel_change_threshold} change in relative value (only from the revised subset):")
+    cli_li(num_percent(rel_change, n_real_revised))
+    na_rel_change <- sum(is.na(real_revisions$max_rel_change)) # nolint: object_usage_linter
+    cli_inform("{units(quick_revision)} until over {percent_final_value} percent of the final value:")
+    difftime_summary(revision_behavior[[glue::glue("time_to_pct_final")]])
+    abs_change <- sum( # nolint: object_usage_linter
+      real_revisions$max_change >
+        abs_change_threshold
+    ) # nolint: object_usage_linter
+    cli_inform("Change by more than {abs_change_threshold} in actual value (when revised):")
+    cli_li(num_percent(abs_change, n_real_revised))
+  }
+  return(revision_behavior)
```

Comment:
```suggestion
  return(invisible(revision_behavior))
```
-----
idea: we might also want to make this an S3-classed thing with a `print` that will print the summary + a pointer to use `as_tibble()` to get the underlying data.

Reply:
I dunno, I personally find printing the full thing under if you haven't assigned it to a variable to be somewhat useful, if maybe a bit cluttered.

Reply:
Sure.  Might consider then:
- If printing the summary, also explicitly print `revision_behavior`, but with a little explanatory blurb beforehand
- Returning invisibly (to not double-print it)